src/release_artifacts_resources/ios/cdk/cdk/credential_rotation/lambda_functions/src/source_data_generator/aws_credential_rotator.py

```diff
@@ -0,0 +1,96 @@
+import boto3
+import os
+from typing import Dict, Tuple
+from src.utils.retry import retry
+from time import sleep
+from datetime import datetime
+
+DEFAULT_REGION = "us-west-2"
+
+# Make sure this is at least 2 hours longer than the key rotation schedule set up in the CFn.
+# Currently, "2 hours" is an arbitrary value based on the fact that a few of our deployments
+# take ~2 hour to run, so we want to leave sufficient time for executing the tests in cases where a
+# test is invoked with session credentials created with access keys that are rotated immediately
+# thereafter.
+SESSION_DURATION_SECONDS = 14400  # 4 hours
+
+# Provided by Lambda
+REGION = os.environ.get("AWS_REGION", DEFAULT_REGION)
+
+def generate_session_credentials(
+    configuration: map, 
+    destination_mapping: list, 
```

Comment:
Let's not have the mapping as part of this logic; the logic to generate & return temporary session credentials doesn't need to know that they'll be mapped to different values elsewhere